Failures just with Android 5.1 (SDK 22), audioplayers 0.18.2

My app started to have failures ONLY on Android 5.1 (SDK 22) after I updated the plugin to 0.18.2

Before version 0.18.2 I used version 0.18.0, so I can't say that it doesn't happen in 0.18.1 too.

Kotlin Version:
ext.kotlin_version = '1.4.31'

**Log Errors**

java.lang.NoClassDefFoundError: 
  at xyz.luan.audioplayers.WrappedMediaPlayer.setUrl (WrappedMediaPlayer.kt:42)
  at xyz.luan.audioplayers.AudioplayersPlugin.handleMethodCall (AudioplayersPlugin.kt:59)
  at xyz.luan.audioplayers.AudioplayersPlugin.onMethodCall (AudioplayersPlugin.kt:34)
  at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage (MethodChannel.java:233)
  at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart (DartMessenger.java:85)
  at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage (FlutterJNI.java:818)
  at android.os.MessageQueue.nativePollOnce (Native Method)
  at android.os.MessageQueue.next (MessageQueue.java:143)
  at android.os.Looper.loop (Looper.java:130)
  at android.app.ActivityThread.main (ActivityThread.java:6946)
  at java.lang.reflect.Method.invoke (Native Method)
  at java.lang.reflect.Method.invoke (Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1404)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1199)
Caused by: java.lang.ClassNotFoundException: 
  at dalvik.system.BaseDexClassLoader.findClass (BaseDexClassLoader.java:56)
  at java.lang.ClassLoader.loadClass (ClassLoader.java:511)
  at java.lang.ClassLoader.loadClass (ClassLoader.java:469)


**Screenshots**
Devices:
![image](https://user-images.githubusercontent.com/10541221/113271700-2d313b80-92b1-11eb-8a1e-3aa6c2845d6a.png)

Android Version:
![image](https://user-images.githubusercontent.com/10541221/113273861-6ff41300-92b3-11eb-8b86-738ec92d1338.png)

The same thing with Android 5.0 with this error but it does not matter what version I used: 

```dart
E/AndroidRuntime(29426): FATAL EXCEPTION: main
E/AndroidRuntime(29426): Process: com.example.feedTest, PID: 29426
E/AndroidRuntime(29426): java.lang.NoClassDefFoundError: Failed resolution of: Landroid/media/MediaDataSource;
E/AndroidRuntime(29426): 	at xyz.luan.audioplayers.WrappedMediaPlayer.setUrl(WrappedMediaPlayer.kt:42)
E/AndroidRuntime(29426): 	at xyz.luan.audioplayers.AudioplayersPlugin.handleMethodCall(AudioplayersPlugin.kt:59)
E/AndroidRuntime(29426): 	at xyz.luan.audioplayers.AudioplayersPlugin.onMethodCall(AudioplayersPlugin.kt:34)
E/AndroidRuntime(29426): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
E/AndroidRuntime(29426): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/AndroidRuntime(29426): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:818)
E/AndroidRuntime(29426): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/AndroidRuntime(29426): 	at android.os.MessageQueue.next(MessageQueue.java:143)
E/AndroidRuntime(29426): 	at android.os.Looper.loop(Looper.java:122)
E/AndroidRuntime(29426): 	at android.app.ActivityThread.main(ActivityThread.java:5264)
E/AndroidRuntime(29426): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(29426): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(29426): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:900)
E/AndroidRuntime(29426): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:695)
E/AndroidRuntime(29426): Caused by: java.lang.ClassNotFoundException: Didn't find class "android.media.MediaDataSource" on path: DexPathList[[zip file "/data/app/com.example.feedTest-2/base.apk"],nativeLibraryDirectories=[/data/app/com.example.feedTest-2/lib/x86, /vendor/lib, /system/lib]]
E/AndroidRuntime(29426): 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
E/AndroidRuntime(29426): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
E/AndroidRuntime(29426): 	at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
E/AndroidRuntime(29426): 	... 14 more
E/AndroidRuntime(29426): 	Suppressed: java.lang.ClassNotFoundException: android.media.MediaDataSource
E/AndroidRuntime(29426): 		at java.lang.Class.classForName(Native Method)
E/AndroidRuntime(29426): 		at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
E/AndroidRuntime(29426): 		at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
E/AndroidRuntime(29426): 		at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
E/AndroidRuntime(29426): 		... 15 more
E/AndroidRuntime(29426): 	Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,7 @@
 ## [next]
 - Refactor Notifications code (small breaking changes)
 - AudioCache for web
+- Fixing basic features for Android lower than API 23
 
 ## 0.18.3
 - Fix Float vs Double mixup on Swift that prevent non-integer values for volume/playback

--- a/android/src/main/kotlin/xyz/luan/audioplayers/WrappedMediaPlayer.kt
+++ b/android/src/main/kotlin/xyz/luan/audioplayers/WrappedMediaPlayer.kt
@@ -52,6 +52,8 @@ class WrappedMediaPlayer internal constructor(
                 player.setDataSource(mediaDataSource)
                 preparePlayer(player)
             }
+        } else {
+            throw RuntimeException("setDataSource is only available on API >= 23");
         }
     }
 

--- a/android/src/main/kotlin/xyz/luan/audioplayers/WrappedMediaPlayer.kt
+++ b/android/src/main/kotlin/xyz/luan/audioplayers/WrappedMediaPlayer.kt
@@ -38,16 +38,20 @@ class WrappedMediaPlayer internal constructor(
             preparePlayer(player)
         }
 
-        // Dispose of any old data buffer array, if we are now playing from another source.
-        dataSource = null
+        if (Build.VERSION.SDK_INT >= 23) {
+            // Dispose of any old data buffer array, if we are now playing from another source.
+            dataSource = null
+        }
     }
 
     override fun setDataSource(mediaDataSource: MediaDataSource?) {
-        if (!objectEquals(dataSource, mediaDataSource)) {
-            dataSource = mediaDataSource
-            val player = getOrCreatePlayer()
-            player.setDataSource(mediaDataSource)
-            preparePlayer(player)
+        if (Build.VERSION.SDK_INT >= 23) {
+            if (!objectEquals(dataSource, mediaDataSource)) {
+                dataSource = mediaDataSource
+                val player = getOrCreatePlayer()
+                player.setDataSource(mediaDataSource)
+                preparePlayer(player)
+            }
         }
     }
 
@@ -109,7 +113,9 @@ class WrappedMediaPlayer internal constructor(
         this.rate = rate.toFloat()
 
         val player = this.player ?: return
-        player.playbackParams = player.playbackParams.setSpeed(this.rate)
+        if (Build.VERSION.SDK_INT >= 23) {
+            player.playbackParams = player.playbackParams.setSpeed(this.rate)
+        }
     }
 
     override fun configAttributes(respectSilence: Boolean, stayAwake: Boolean, duckAudio: Boolean) {
@@ -205,7 +211,7 @@ class WrappedMediaPlayer internal constructor(
             if (released || currentPlayer == null) {
                 released = false
                 player = createPlayer().also {
-                    if (dataSource != null) {
+                    if (Build.VERSION.SDK_INT >= 23 && dataSource != null) {
                         it.setDataSource(dataSource)
                     } else {
                         it.setDataSource(url)

--- a/feature_parity_table.md
+++ b/feature_parity_table.md
@@ -31,7 +31,7 @@ Note: LLM means Low Latency Mode.
         <tr><td>volume</td><td>yes</td><td>yes</td><td>yes</td><td>yes</td></tr>
         <tr><td>seek</td><td>yes</td><td>yes</td><td>yes</td><td>not yet</td></tr>
         <tr><td colspan="5"><strong>Advanced Audio Control Commands</strong></td></tr>
-        <tr><td>playback rate</td><td>yes</td><td>yes</td><td>yes</td><td>not yet</td></tr>
+        <tr><td>playback rate</td><td>SDK >= 23</td><td>yes</td><td>yes</td><td>not yet</td></tr>
         <tr><td>duck audio</td><td>yes (except LLM)</td><td>no</td><td>no</td><td>no</td></tr>
         <tr><td>respect silence</td><td>yes (except LLM)</td><td>yes</td><td>no</td><td>no</td></tr>
         <tr><td>stay awake</td><td>yes (except LLM)</td><td>yes</td><td>no</td><td>no</td></tr>

